chain/client/src/stateless_validation/chunk_endorsement_tracker.rs

```diff
@@ -23,15 +23,29 @@ pub struct ChunkEndorsementTracker {
     /// This is keyed on chunk_hash and account_id of validator to avoid duplicates.
     /// Chunk endorsements would later be used as a part of block production.
     chunk_endorsements: SyncLruCache<ChunkHash, HashMap<AccountId, ChunkEndorsement>>,
+    /// We store chunk endorsements to be processed later because we did not have
+    /// chunks ready at the time we received that endorsements from validators.
+    /// This is keyed on chunk_hash and account_id of validator to avoid duplicates.
+    pending_chunk_endorsements: SyncLruCache<ChunkHash, HashMap<AccountId, ChunkEndorsement>>,
 }
 
 impl Client {
     pub fn process_chunk_endorsement(
         &mut self,
         endorsement: ChunkEndorsement,
     ) -> Result<(), Error> {
-        let chunk_header = self.chain.get_chunk(endorsement.chunk_hash())?.cloned_header();
-        self.chunk_endorsement_tracker.process_chunk_endorsement(&chunk_header, endorsement)
+        // We should not need whole chunk ready here, we only need chunk header.
+        match self.chain.get_chunk(endorsement.chunk_hash()) {
+            Ok(chunk) => {
+                let chunk_header = &chunk.cloned_header();
+                self.chunk_endorsement_tracker
+                    .process_chunk_endorsement(endorsement, Some(chunk_header))
+            }
+            Err(_) => {
+                tracing::debug!(target: "stateless_validation", ?endorsement, "Endorsement arrived before chunk.");
```

Comment:
nit 1: Can you match the exact error and treat any other as an actual error? 
nit 2: Can you add the chunk hash to the debug log? It helps debugging e.g. when grepping for the chunk hash.

Reply:
Chunk hash is part of endorsement so it is already debug printed with `?endorsement`